packages/ai/src/vercel/middleware.ts

```diff
@@ -152,25 +153,59 @@ const mapVercelPrompt = (messages: LanguageModelV2Prompt): PostHogInput[] => {
   return inputs
 }
 
-const mapVercelOutput = (result: any): PostHogInput[] => {
-  const content: any[] = []
-
-  if (result.text) {
-    content.push({ type: 'text', text: truncate(result.text) })
-  }
-
-  if (result.toolCalls && Array.isArray(result.toolCalls)) {
-    for (const toolCall of result.toolCalls) {
-      content.push({
-        type: 'function',
-        id: toolCall.toolCallId,
+const mapVercelOutput = (result: LanguageModelV2Content[]): PostHogInput[] => {
+  const content: any[] = result.map((item) => {
+    if (item.type === 'text') {
+      return { type: 'text', text: truncate(item.text) }
+    }
+    if (item.type === 'tool-call') {
+      return {
+        type: 'tool-call',
```

Comment:
We made tool calls all have a type "function" so that they can be parsed in the new flow that takes an array in the content property. I'm not sure how this is going to be displayed.